### PR TITLE
add ISRC support

### LINKED ISSUE #1: Support reading/submitting ISRCs
Some image formats contain enough information to get isrcs out. See if libmirage interfaces can be used for this, and support submitting them.

### PR DESCRIPTION
**Pull request by JonnyJD**
This prints found ISRCs on stderr, since only the actual disc ID is
given on stdout.

closes #1

### CODE REVIEW

**Comment by JonnyJD** [comment]
For a quick test I found a .cue which comes along with some flac files and added an ISRC (taken from somewhere else) manually:

```
...
FILE "04 - Camouflage - Your Skinhead Is the Dream.flac" WAVE
  TRACK 04 AUDIO
    TITLE "Your Skinhead Is the Dream"
    PERFORMER "Camouflage"
    INDEX 00 00:00:00
    INDEX 01 00:03:38
    ISRC USSM19501455
...
```

which leads to:

```
...
Full TOC: 1 10 177075 183 21015 42608 59848 82473 105128 124140 141768 158558 174340
FSJEHGsSD0YoJL1ab3fFjsToDvg-
FreeDB: 8509370a
http://mm.musicbrainz.org/bare/cdlookup.html?id=FSJEHGsSD0YoJL1ab3fFjsToDvg-&tracks=10&toc=1+10+177075+183+21015+42608+59848+82473+105128+124140+141768+158558+174340
ISRC Track 4: USSM19501455
```

This would make it easy to be used by isrcsubmit as input (having track number and the word ISRC on one line). The actual order of these doesn't matter, but shouldn't be changed at will after this is merged and released.

**Comment by JonnyJD** [comment]
http://cdemu.sourceforge.net/reference/MirageTrack.html#mirage-track-get-isrc :

> The string belongs to the object and should not be modified.

This probably means we shouldn't use the string after calling `g_object_unref(track)`.

**Comment by kepstin** [comment]
Yeah, it would probably be best to simply print off the ISRC for each track as that track is read.

There's a lot of debugging output in image-id that is really unnecessary; I would recommend printing out the ISRCs on stdout in order to distinguish them from the debugging output. (And I should probably print the final disc ids on stdout as well.)

**Comment by JonnyJD** [comment]
Well, I thought there are scripts that expect use image id like this:

```
disc_id = `./image_id $cuefile 2> /dev/null`
```

Using stdout for anything other than the disc ID would break this.

**Comment by JonnyJD** [comment]
I moved to using strcpy to "our" memory, but still print the ISRCs at the end (to stderr).

**Comment by JonnyJD** [comment]
Even if moving to stdout. The id should be the first thing that is displayed.

An alternative (as a new "API") would be to prepend `Disc ID:` to the disc ID output so it can easily parsed (grep+awk).

**Comment by JonnyJD** [comment]
I merged the changes from master. Now using stdout for ISRCs. I still save them in an array to display them after the disc ID is displayed.

File: image_id.c [hunk 1]
```diff
@@ -93,6 +93,7 @@ static bool process_disc(MirageDisc *disc, DiscId *discid) {
 
 	int first, last;
 	int offsets[100] = {0};
+	char isrcs[100][12+1] = {{0}};
```
Comment (kepstin) [inline image_id.c:96]: Having an array with 100 entries here almost seems a bit silly when a CD can only have 99 tracks; but this does simplify the code in other places by allowing direct index by (1-based) track number. I suppose that's ok.

File: image_id.c [hunk 2]
```diff
@@ -175,6 +176,12 @@ static bool process_disc(MirageDisc *disc, DiscId *discid) {
 			}
 
 			offsets[track_num] = track_start + track_start_sector + offset;
+			if (mirage_track_get_isrc(track) != NULL) {
+				strncpy(isrcs[track_num],
+					(char *) mirage_track_get_isrc(track),
```
Comment (kepstin) [inline image_id.c]: The cast on this line loses the 'constness' of the string returned by `mirage_track_get_isrc`, which is bad form. I think that the `const gchar *` and `const char *` types are compatible, so you should be able to get by without any cast here at all.

Comment (JonnyJD) [inline image_id.c]: Yes. That got left over from debugging when I forgot to check for NULL. I removed the typecast.

File: image_id.c (hunk 1 above)
Comment (JonnyJD) [inline image_id.c:96]: I'd rather use the same indexing as `offsets[]`. Libdiscid also uses this indexing internally for isrcs.

**Comment by kepstin** [comment]
I had already merged in a slightly older version of this branch for testing with my own fix for the char cast, I ended up just pushing that.

**Comment by JonnyJD** [comment]
Thanks.

**Pull request closed without merging**